sdk/core/azure-core/inc/azure/core/credentials/credentials.hpp

```diff
@@ -41,13 +43,38 @@ namespace Azure { namespace Core { namespace Credentials {
   /**
    * @brief Context for getting token.
    */
-  struct TokenRequestContext final
-  {
+  class TokenRequestContext final {
+  public:
+    /**
+     * @brief Default constructor
+     *
+     */
+    TokenRequestContext() = default;
+    /**
+     * @brief Scopes constructor
+     *
+     */
+    TokenRequestContext(std::vector<std::string> scopes) : Scopes(std::move(scopes)){};
```

Comment:
Take a look to: https://github.com/Azure/azure-sdk-for-cpp/pull/3367#discussion_r811522693

We talked about this in a C++ group meeting.  In summary, `const&` will always make a copy, while `non-const&` will depend on lvalue or rvalue

Reply:
The link to a comment doesn't work as I would like,  here's a copy paste:

```
I think we've talked about this several times in different occasions.

Basically, since we're defining a constructor here, we don't know what kind of value callers are going to pass, it could be an lvalue or an rvalue.

In the case of Foo(std::string f): m_f(std::move(f)):

a. if parameter is lvalue, we make a copy in f then move from f to m_f.
b. if parameter is rvalue, we move from that rvalue to f, then move to m_f. No copy is made!

But in the case of Foo(const std::string& f): m_f(f), we make a copy regardless the parameter is lvalue or rvalue.

I'm not the only one that prefers this style. I think Casey is with me. https://github.com/Azure/azure-sdk-for-cpp/pull/2595#discussion_r671440673

There are also a lot of discussion on stackoverflow, like [this](https://stackoverflow.com/questions/51705967/advantages-of-pass-by-value-and-stdmove-over-pass-by-reference) one
```

Reply:
Fair enough.

Reply:
done